pii_recognition/registers/registry_test.py

```diff
@@ -0,0 +1,29 @@
+from unittest.mock import patch
+
+from pytest import raises
+
+from .registry import Registry
+
+
+@patch.object(Registry, attribute="__abstractmethods__", new=set())
+def test_Registry():
+    def toy_func():
+        pass
+
+    class ToyClass:
+        pass
+
+    toy_assignment = ""
+
+    actual = Registry()  # type: ignore
+    actual.add_item(toy_func)
+    assert isinstance(actual, dict)
+    assert actual["toy_func"] == toy_func
+
+    actual.add_item(ToyClass)
+    assert isinstance(actual, dict)
+    assert actual["ToyClass"] == ToyClass
+
+    with raises(AttributeError) as err:
+        actual.add_item(toy_assignment)
+    assert str(err.value) == "'str' object has no attribute '__name__'"
```

Comment:
can we have a more meaningful error message here? something like `the registered item should be either a class or function`?

Reply:
I thought about that as well. Initially I planned to set restrictions to only class and function, but found out not all functions have `__name__` attribute according to a SO. So, instead I stick with the built-in error message from `getattr`.

More elaborate message example would be `The registered item str is not supported as str does not has attribute __name__, you need to register the iterm as key-value pair. `, what you think?

Reply:
sounds good!

Reply:
Seems strange to me the only requirement to be added to the registry is to have a `__name__` attribute. What benefit do you really get from this than using a simple dictionary?

Reply:
The main reason is to separate registries for different collections when creating child class, i.e., `TokeniserRegistry`, `Detokeniser` and `RecogniserRegistry`.

Beside of that, a couple of other methods I was thinking to add to this class, would like to know what you think about the below
1. Modify to `add_item(self, item: T, name: Optional[str])` which using name as an optional argument, that is, you can assign a name to an item, if not will use `__name__`.
2. Add a method `get_instance(self, name, **kwargs)`. When called it will return a fully initialised object. Otherwise, users have to complete the initialisation, always.
3. Set restrictions -- can only register `class` objects no function objects or anything else. For now, tokenisers and detokenisers implemented in functions since everything is still basic. Once they become complex, i.e., requiring config, function is not enough, not flexible and consistent.